guardian/lib/slack/model/slack_message.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:equatable/equatable.dart';
+import 'package:guardian/slack/model/slack_section_block.dart';
+
+/// A class representing Slack message to be sent.
+class SlackMessage extends Equatable {
+  /// The main body text of the message.
+  ///
+  /// It can be formatted as plain text, or with markdown.
+  /// If [blocks] is presented, this is used as a fallback string to display
+  /// in notifications.
+  final String text;
+
+  /// An array of layout sections.
+  ///
+  /// Slack Block Kit is described here https://api.slack.com/block-kit.
+  /// Only section blocks are currently implemented.
+  final List<SlackSectionBlock> blocks;
+
+  /// Creates an instance of Slack message.
+  ///
+  /// If [blocks] is presented, [text] is not enforced as required, however it
+  /// is highly recommended to be included as the aforementioned fallback.
+  const SlackMessage({
+    this.text,
+    this.blocks,
+  });
```

Comment:
Should this class constructor be more strict with asserts? Making sure that text or blocks provided?